tests/bluetooth/bsim_bt/bsim_test_audio/src/has_client_test.c

```diff
@@ -0,0 +1,279 @@
+/*
+ * Copyright (c) 2022 Codecoup
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifdef CONFIG_BT_HAS_CLIENT
+#include "bluetooth/audio/has.h"
+#include "common.h"
+
+extern enum bst_result_t bst_result;
+
+static volatile bool g_bt_init;
+static volatile bool g_is_connected;
+static volatile bool g_mtu_exchanged;
+static volatile bool g_has_discovered;
+static volatile bool g_active_preset_notified;
+static volatile bool g_preset_read;
+
+static struct bt_conn *g_conn;
+static struct bt_has *g_has;
+static volatile uint8_t g_active_preset_id;
+
+static void discover_cb(struct bt_conn *conn, struct bt_has *has, bt_has_hearing_aid_type_t type)
+{
+	if (!has) {
+		FAIL("Failed to discover HAS\n");
+		return;
+	}
+
+	printk("HAS discovered (type %d)\n", type);
+
+	g_has = has;
+	g_has_discovered = true;
+}
+
+static void active_preset_cb(struct bt_has *has, int err, uint8_t id)
+{
+	if (err != 0) {
+		FAIL("Failed to get active preset (err %d)\n", err);
+		return;
+	}
+
+	g_active_preset_id = id;
+	g_active_preset_notified = true;
+}
+
+static void preset_cb(struct bt_has *has, int err, uint8_t id, uint8_t properties, const char *name)
+{
+	printk("id %d properties 0x%02x name %s\n", id, properties, name);
+}
+
+static void preset_read_complete_cb(struct bt_has *has, int err)
+{
+	g_preset_read = true;
+}
+
+static struct bt_has_client_cb has_cb = {
+	.discover = discover_cb,
+	.active_preset = active_preset_cb,
+	.preset = preset_cb,
+	.preset_read_complete = preset_read_complete_cb,
+};
+
+static void connected(struct bt_conn *conn, uint8_t err)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	if (err > 0) {
+		FAIL("Failed to connect to %s (err %u)\n", addr, err);
+		return;
+	}
+	printk("Connected to %s\n", addr);
+	g_conn = conn;
+	g_is_connected = true;
+}
+
+BT_CONN_CB_DEFINE(conn_callbacks) = {
+	.connected = connected,
+	.disconnected = disconnected,
+};
+
+static void bt_ready(int err)
+{
+	if (err != 0) {
+		FAIL("Bluetooth discover failed (err %d)\n", err);
+		return;
+	}
+
+	g_bt_init = true;
+}
```

Comment:
You can omit the callback to `bt_enable` to make it synchronous and avoid this function and `g_bt_init`